AEPAnalytics/Tests/AnalyticsTests.swift

```diff
@@ -386,4 +399,120 @@ class AnalyticsAPITests: XCTestCase {
 
         XCTAssertTrue(NSDictionary(dictionary: edgeEvent.data!).isEqual(to: expectedData))
     }
+
+    func testIds_areSentIfPresentInDatastore() {
+        dataStore.set(key: AnalyticsConstants.DataStoreKeys.VID, value: "testvid")
+        dataStore.set(key: AnalyticsConstants.DataStoreKeys.AID, value: "testaid")
+
+        let trackData: [String: Any] = [
+            AnalyticsConstants.EventDataKeys.TRACK_INTERNAL: true,
+            CoreConstants.Keys.ACTION : "action",
+            CoreConstants.Keys.CONTEXT_DATA : [:]
+        ]
+        let trackStateEvent = Event(name: "Generic track event", type: EventType.genericTrack, source: EventSource.requestContent, data: trackData)
+
+        mockRuntime.simulateSharedState(extensionName: AnalyticsConstants.Configuration.SHARED_STATE_NAME, event: trackStateEvent, data:
+            ([AnalyticsConstants.Configuration.GLOBAL_CONFIG_PRIVACY : PrivacyStatus.unknown.rawValue], .set)
+        )
+        mockRuntime.simulateComingEvent(event: trackStateEvent)
+
+        XCTAssertEqual(mockRuntime.dispatchedEvents.count, 1)
+        guard let eventDataDict = mockRuntime.dispatchedEvents[0].data else {
+            XCTFail("Failed to convert event data to [String: Any]")
+            return
+        }
```

Comment:
the tests can be ignored from code coverage by adding a codecov.yml file with the tests directories added to ignore. see https://github.com/adobe/aepsdk-audience-ios/blob/main/codecov.yml#L20

Reply:
I thought `./AEPAnalytics/Tests/**/*` removes every file from test directory. Guess it does not, i will update the ignore file.